### PR TITLE
CMake fixes

### PR DESCRIPTION
**Pull request by diizzyy**
Make lib(WebP) and OpenJPEG support optional
Report correct version in generated .pc file
Populate Libs.private in .pc file
Properly define library version
Install imagetops for consistency with Autotools build

### CODE REVIEW

**Comment by diizzyy** [comment]
You're free to change whether these should be enabled or disabled by default

**Comment by zdenop** [comment]
What is the rationale behind this? Why do you turn them off by default?

**Comment by diizzyy** [comment]
As I said, feel free to use whatever option as you set fit as default. The main point is to make sure they aren't pulled in unintentionally.

**Comment by diizzyy** [comment]
Remaning issues:
Library symlinks are a bit weird/inconsistent to autotools

```
lrwxr-xr-x  1 root  wheel       21 Dec  4 15:44 libleptonica.so -> libleptonica.so.5.4.0
-rwxr-xr-x  1 root  wheel  2849560 Dec  4 15:44 libleptonica.so.1.82.0
lrwxr-xr-x  1 root  wheel       22 Dec  4 15:44 libleptonica.so.5.4.0 -> libleptonica.so.1.82.0
```

**Comment by DanBloomberg** [comment]
@egorpugin @stweil @zdenop
There are 5 relatively minor changes to cmake in this PR.
Do you have any issues with them?

Two questions:
(1) I don't know anything about private libs in cmake.  Are they useful?  Suggest that if they are adopted, the default be ON for jp2k and webp.
(2) Is there any reason for the use of the version number (e.g., 1.82.0) instead of the so number in the name of the library?

**Comment by diizzyy** [comment]
Private.libs matters because if you have a downstream user that looks for the pc file (pkgconfig) it'll break.
As for so number this should be consistent with autotools

File: CMakeLists.txt [hunk 1]
```diff
@@ -55,70 +55,93 @@ else()
...
+    option(OPENJPEG_SUPPORT "Build with OpenJPEG support" OFF)
 endif()
 
 set(leptonica_INSTALL_CMAKE_DIR "lib/cmake/leptonica"
     CACHE STRING "Install destination for CMake package files")
 
 if(NOT SW_BUILD)
     find_package(GIF)
+        if(GIF_FOUND)
+            set(libs_private "-lgif")
```
Comment (stweil) [inline CMakeLists.txt:68]:
```suggestion
            set(libs_private "${libs_private} -lgif")
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -55,70 +55,93 @@ else()
...
+    option(LIBWEBP_SUPPORT "Build with WebP support" OFF)
+    option(OPENJPEG_SUPPORT "Build with OpenJPEG support" OFF)
 endif()
 
 set(leptonica_INSTALL_CMAKE_DIR "lib/cmake/leptonica"
     CACHE STRING "Install destination for CMake package files")
 
 if(NOT SW_BUILD)
     find_package(GIF)
+        if(GIF_FOUND)
```
Comment (stweil) [inline CMakeLists.txt:67]: Why are this and the other `if` blocks not on the same indentation level as `find_package`?

File: CMakeLists.txt [hunk 3]
```diff
@@ -55,70 +55,93 @@ else()
 endif()
 if(UNIX)
     option(SYM_LINK "Create symlink leptonica -> lept on UNIX" OFF)
+    option(LIBWEBP_SUPPORT "Build with WebP support" OFF)
+    option(OPENJPEG_SUPPORT "Build with OpenJPEG support" OFF)
```
Comment (stweil) [inline CMakeLists.txt:59]:
```suggestion
    option(LIBWEBP_SUPPORT "Build with WebP support" ON)
    option(OPENJPEG_SUPPORT "Build with OpenJPEG support" ON)
```

**Review by stweil — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (diizzyy) [reply CMakeLists.txt:68]: Is that correct since it's initializing the variable?

File: CMakeLists.txt (hunk 2 above)
Comment (diizzyy) [reply CMakeLists.txt:67]: Good catch, I'll correct that

File: CMakeLists.txt (hunk 1 above)
Comment (stweil) [reply CMakeLists.txt:68]: It isn't, unless `GIF_FOUND` was true. Any of the following `set` statements could initialize it if all preceding `if` blocks are false.

Comment (stweil) [reply CMakeLists.txt:68]: You could initialize `libs_private` before the first `if` block to an empty string if necessary. The resulting value of `libs_private` can have a leading space. I don't know whether that is a problem.

**Comment by diizzyy** [comment]
@stweil 
Fixed, thanks for reviewing!

**Comment by diizzyy** [comment]
Anything else that needs to be done and/or adjusted?

**Comment by diizzyy** [comment]
Friendly ping

**Comment by DanBloomberg** [comment]
Daniel, approving your PR is above my pay grade.  The only build system I maintain is the hand-tweaked makefiles.
My impression of the changes:
Make lib(WebP) and OpenJPEG support optional:       ok with default ON
Report correct version in generated .pc file:                  seems reasonable
Populate Libs.private in .pc file:                                     seems reasonable
Properly define library version:                                      good to follow standards
Install imagetops for consistency with Autotools build:  ok

Anyone want to approve or suggest changes?

**Comment by stweil** [comment]
> I don't know anything about private libs in cmake. Are they useful?

`lept.pc` on Debian has `Libs.private: -lz -lpng16 -lz -ljpeg -lopenjp2 -lgif -ltiff -lwebp -lwebpmux -lwebp` (note that `-lwebp` is unnecessarily duplicated), so adding the same libraries in cmake builds looks correct.

I'm afraid I cannot say much to the remaining changes as I don't think that I am an expert for cmake. @egorpugin and @zdenop contributed most to `CMakeLists.txt`

**Comment by DanBloomberg** [comment]
@diizzyy 

Please look at
   https://github.com/tesseract-ocr/tesseract/issues/3738

While trying to resolve that issue, we're brought back to the various library naming issues.

**Pull request merged**